lib/streamlit/DeltaGenerator.py

```diff
@@ -256,19 +273,23 @@ def value_or_dg(value, dg):
             if elementHeight is not None:
                 msg.metadata.element_dimension_spec.height = elementHeight
 
+
         # "Null" delta generators (those without queues), don't send anything.
         if self._enqueue is None:
             return value_or_dg(rv, self)
 
         # Figure out if we need to create a new ID for this element.
         if self._is_root:
             output_dg = DeltaGenerator(
-                self._enqueue, msg.metadata.delta_id, is_root=False
-            )
+                self._enqueue, msg.metadata.delta_id, delta_type, last_index,
```

Comment:
Now that `DeltaGenerator.__init__` has so many arguments it's best to initialize it using only keyword arguments, so it's less likely we'll break things in the future:

```
 DeltaGenerator(enqueue=self._enqueue, id=..., delta_type=..., last_index=..., is_root=False)
```